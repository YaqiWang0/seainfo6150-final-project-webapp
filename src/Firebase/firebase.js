import app from 'firebase/app'
import 'firebase/auth'
import 'firebase/database'

const config = {
    apiKey: "AIzaSyCBcPf2qCe4gcUgEeZBpdowfPs_l5HDNsI",
    authDomain: "fall2020-6bd75.firebaseapp.com",
    databaseURL: "https://fall2020-6bd75.firebaseio.com",
    projectId: "fall2020-6bd75",
    storageBucket: "fall2020-6bd75.appspot.com",
    messagingSenderId: "671808993553",
    appId: "1:671808993553:web:87adf307e12583f441f4b0",
    measurementId: "G-79CEYSDSE9"
};

class Firebase {
    constructor() {
        app.initializeApp(config);

        this.auth = app.auth();
        this.db = app.database();
    }


    doCreateUserWithEmailAndPassword = (email, password) =>
        this.auth.createUserWithEmailAndPassword(email, password);

    doSignInWithEmailAndPassword = (email, password) =>
        this.auth.signInWithEmailAndPassword(email, password);

    doSignOut = () => this.auth.signOut();

    user = (uid) => this.db.ref(`users/${uid}`);

    users = () => this.db.ref('users');

}

export default Firebase;
