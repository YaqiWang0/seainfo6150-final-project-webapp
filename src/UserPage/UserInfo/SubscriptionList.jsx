import React, {useState, useEffect}from "react"
import { isEmpty } from "lodash";

import style from "./SubscriptionList.module.css"
import {useHistory} from "react-router-dom";

const SubscriptionList = (props) => {
    const [fetchedData, setFetchedData] = useState({});
    let history = useHistory();
    useEffect(() => {
        const fetchData = async () => {

            let variable = "";
            for(var i = 0; i < props.sub.length - 1; i++) {
                variable = variable + props.sub[i] + ","
            }
            variable = variable + props.sub[props.sub.length - 1];

            const response = await fetch(`https://financialmodelingprep.com/api/v3/quote/${variable}?apikey=56f622a93b55019c5a0875058fc58f13`);
            const responseJson = await response.json();
            setFetchedData(responseJson);
        };
        if(isEmpty(fetchedData)) {
            fetchData()
        }
    }, [fetchedData]);

    const click = (index) => {
        // let subs = [...subscribe];
        // let item = {...subs[index]};
        // item.isSubscribed = !item.isSubscribed;
        // subs[index] = item;
        // setSubscribe(subs);
        // console.log(subscribe[`${index}`])
        setFetchedData(fetchedData.filter((e, i) => i !== index))
    }

    const viewDetail = async (index) => {
        const response = await fetch(`https://financialmodelingprep.com/api/v3/profile/${props.sub[index]}?apikey=56f622a93b55019c5a0875058fc58f13`);
        const responseJson = await response.json();
        const newsResponse = await fetch(`https://financialmodelingprep.com/api/v3/stock_news?tickers=${props.sub[index]}&limit=10&apikey=56f622a93b55019c5a0875058fc58f13`);
        const newsResponseJson = await newsResponse.json();
        history.push(
            "/company",
            {
                info: Object.values(responseJson),
                news: newsResponseJson,
                login: props.login
            }
        )
    }

    return (
        <ul className={style.container}>
        {
            Object.values(fetchedData).map((item, index) => (
                <li key={index} className={style.textInfo}>
                    <div className={style.title} onClick={() => viewDetail(index)}>{item.name}</div>
                    <div className={style.sub}>
                        <div className={style.stockPriceBlock}>
                            <div className={item.change >= 0 ? style.price_red : style.price_green}>{`$${item.price}`}</div>
                            <div className={item.change >= 0 ? style.stock_red : style.stock_green}>{item.change >=0 ? `(+ ${item.change})` : `( ${item.change})`}</div>
                        </div>
                        <div onClick={ () => click(index)} className={style.heartSub}></div>
                    </div>
                </li>
            ), this)
        }
        </ul>
    )
}

export default SubscriptionList