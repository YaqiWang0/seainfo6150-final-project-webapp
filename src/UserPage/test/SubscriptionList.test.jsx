import React from "react";

import Firebase from "../../Firebase";
import SubscriptionList from "../UserInfo/SubscriptionList";

describe("SubscriptionList test", () => {

    const login = {
        login: true,
        userId: "NOFjyl43m1Xy3rB4umSRERq8M2z1",
        email: "123456@test.com"
    }

    it('render successfully', () => {
        const firebase = new Firebase();
        const {container} = render(
            <SubscriptionList login = {login} firebase = {firebase}/>
        );
        expect(container).toMatchSnapshot();
    })
})