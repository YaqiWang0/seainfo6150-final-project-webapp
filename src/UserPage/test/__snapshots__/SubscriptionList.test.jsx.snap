// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubscriptionList test render successfully 1`] = `
<div>
  <ul
    class="container"
  />
</div>
`;
